### ISSUE TITLE
Support descriptions for services, methods, and fields to show the description in the UI

### ISSUE THREAD

**Issue by flymedllva** [connectrpc/grpcreflect-go#78]
In this [PR](https://github.com/fullstorydev/grpcui/pull/166) for UI, support for describing protobuf entities has been implemented. 

This works well if you use local `.proto`.
`grpcui -proto test.proto -plaintext localhost:8080`
`description: "// test comment\noptional int64 test_id = 1;\n"`

But when using grpcreflect-go, comments do not appear in the description line
`description: "optional int64 test_id = 1;\n"`

**Comment by jhump** [comment]
> But when using grpcreflect-go, comments do not appear in the description line

@flymedllva, this also happens when using the reflection server implementation in the google.golang.org/grpc module, too. The limitation has nothing to do with this module or package but with the code generated by protoc-gen-go: the embedded descriptors, which are accessed by gRPC server reflection, _do not have source code info_. Comments are stripped from the descriptors before they are embedded into generated code: [link to source](https://github.com/protocolbuffers/protobuf-go/blob/v1.34.2/cmd/protoc-gen-go/internal_gengo/reflect.go#L259). This is actually true of _every_ official Protobuf runtime (for all languages): the descriptors embedded in generated code do not have any comments. This was a choice to reduce generated code size and binary sizes.

Read more in related issues [here](https://github.com/protocolbuffers/protobuf/issues/8728) and [here](https://github.com/bazelbuild/bazel/issues/9337).

In short, it is not possible for this module to support comments for these elements, at least not with `NewStaticReflector`, because the source of the reflection data (the code generated by `protoc-gen-go`) does not contain any of those comments.

For Go, I've written a work-around that uses a [custom proto plugin](https://pkg.go.dev/github.com/jhump/protoreflect@v1.16.0/desc/sourceinfo/cmd/protoc-gen-gosrcinfo) to generate extra code: [github.com/jhump/protoreflect/desc/sourceinfo](https://pkg.go.dev/github.com/jhump/protoreflect/desc/sourceinfo). That package provides registries that you can use with options when calling `grpcreflect.NewReflector` that will restore comments to the descriptors.

**Comment by flymedllva** [comment]
@jhump 

I tried using the plugin to generate and swap registry, it actually worked for me.
Thanks!

Just in case code, for those who are interested:
```go
type staticNames struct {
	names []string
}

func (n *staticNames) Names() []string {
	return n.names
}

reflector := grpcreflect.NewReflector(
	&staticNames{
		names: servicesNames,
	},
	grpcreflect.WithExtensionResolver(sourceinfo.GlobalTypes),
	grpcreflect.WithDescriptorResolver(sourceinfo.GlobalFiles),
)
mux.Handle(grpcreflect.NewHandlerV1(
	reflector,
))
mux.Handle(grpcreflect.NewHandlerV1Alpha(
	reflector,
))
```

Maybe it should be written somewhere in the documentation?

But in general is it a normal practice to do this? Replace the original `google.golang.org/protobuf/reflect/protoregistry`, have you encountered any problems in the future?

**Comment by jhump** [comment]
> But in general is it a normal practice to do this?

No, it is not. In general, people usually simply make do without the comments being available. The main downside is increased code and binary size (which can translate to increased memory usage in the running program).

As far as "replacing" the originals: this isn't really doing that: `protoregistry.GlobalFiles` still exists as it was and is usable by all other reflection tools, unchanged. We're just swapping in a separate registry for the reflection service. I know of no downside to that, other than the extra size/memory as mentioned above.
